### PR TITLE
fix(unify): Use of experimentalStudio should throw a warning that it is removed

### PR DESCRIPTION
**Pull request by davidmunechika**
- Closes https://cypress-io.atlassian.net/browse/UNIFY-1093?atlOrigin=eyJpIjoiZjUxZmExNThlMTFmNDM1NWEzNTk2Mzc5Yzk4Zjc3NDMiLCJwIjoiaiJ9

### User facing changelog
`experimentalStudio` will be removed in 10.0.0. This adds messaging to warn when a user uses this flag to inform them that this has been removed so that they don't think it's a working feature that is broken when we release.

### How has the user experience changed?

Run mode:

Screenshot 2022-03-02 at 8 57 33 AM

Open mode: 

Screenshot 2022-03-08 at 11 08 03 AM

Screenshot 2022-03-07 at 1 59 18 PM (1)

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [x] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/errors/src/errors.ts
Comment (emilyrohrbough) [inline packages/errors/src/errors.ts:1062]: Are we officially releasing studio when we re-introduce it?

Comment (jennifer-shehane) [reply packages/errors/src/errors.ts:1062]: Yah, not completely clear on what the release will look like yet. 

This discussions link needs to be an on link. There should already be an on link for this as we used this in product with the Cypress Studio feedback before.

**Review by jennifer-shehane — DISMISSED** [review]

File: packages/config/lib/options.ts
Comment (emilyrohrbough) [inline packages/config/lib/options.ts:529]: Why is this needed 2x?

Comment (emilyrohrbough) [reply packages/config/lib/options.ts:529]: This option needs to be removed in the options above: https://github.com/cypress-io/cypress/pull/20515/files#diff-3327662b7aa1effb2d97569ca234bf56dbd83bc0664c63c9be165e1923615e65R178

We also need to remove this config option from the types and json schema in cypress/cli

File: packages/config/lib/index.js
Comment (emilyrohrbough) [inline packages/config/lib/index.js:135]: Just curious, are we able to use the existing API used to display the baseUrl warning message in the launchpad?

Comment (davidmunechika) [reply packages/config/lib/index.js:135]: We don't have the baseUrl warning message implemented yet in 10.0. I think Zach Williams is working on that in a different PR right now

Comment (emilyrohrbough) [reply packages/config/lib/index.js:135]: ohhh I thought that had merged

**Review by flotwig — APPROVED** [review]
LGTM code-wise

File: packages/config/lib/options.ts
Comment (davidmunechika) [reply packages/config/lib/options.ts:529]: Screenshot 2022-03-09 at 11 28 58 AM

If I remove it from the breakingOptions, then it won't display a warning in the terminal. The breaking options are what we use in the validation function

Comment (davidmunechika) [reply packages/config/lib/options.ts:529]: The breakingRootOptions allows the warning to show up in open mode as well

File: packages/errors/src/errors.ts
Comment (jennifer-shehane) [inline packages/errors/src/errors.ts:1062]: @davidmunechika Is this link working when you click on it? In GitHub it's adding the period to the url so the link doesn't work, but this may not be a problem in the actual product.

Comment (davidmunechika) [reply packages/errors/src/errors.ts:1062]: Screenshot 2022-03-09 at 12 07 08 PM

Yes, this works for me in the app. It separates the link from the period.

**Comment by lmiller1990** [comment]
Looks like have two approvals, one from @flotwig and one from @jennifer-shehane. just missing the actual :heavy_check_mark: Might be good to get one final check and :heavy_check_mark: since there have been some code changes.

**Comment by flotwig** [comment]
@lmiller1990 @emilyrohrbough should probably review again too, since she had some comments. Generally I try not to merge something, even if it had 2 approvals, if there are unresolved review comments.

**Review by flotwig — DISMISSED** [review]
@davidmunechika is it correct that the Warning header has no background color? Looks a little odd.

**Comment by davidmunechika** [comment]
> @davidmunechika is it correct that the Warning header has no background color? Looks a little odd.

@flotwig No, this is a styling bug in the Warning component itself. However, Zach already has a fix open for this in his PR: https://github.com/cypress-io/cypress/pull/20490

**Comment by lmiller1990** [comment]
@flotwig Agreed - let's wait for a re-review. :+1:

**Review by emilyrohrbough — CHANGES_REQUESTED** [review]
According to the 10.0 Changes document, this should throw an error when this is passed: https://www.notion.so/cypressdx/Changes-for-10-0-0-1dca213837b84d6ea297c2d12c615f3a#fc91c34e28054f9b934a1bc6dbbe53a0

It seems like @brian-mann / @jennifer-shehane / @cowboy have decided this should be a warning? If this is the new path forward, an you update the notion doc?

File: packages/data-context/src/data/ProjectLifecycleManager.ts
Comment (emilyrohrbough) [inline packages/data-context/src/data/ProjectLifecycleManager.ts:599]: Does ctx.onError handle throw this for us?

File: packages/data-context/src/data/ProjectLifecycleManager.ts
Comment (emilyrohrbough) [inline packages/data-context/src/data/ProjectLifecycleManager.ts:594]: not sure this return is necessary since we created this func inline., however we can return `return error` since you created it on line 590.

File: packages/errors/src/errors.ts
Comment (emilyrohrbough) [inline packages/errors/src/errors.ts:1061]: Most errors we include the cypress version this changed in. Do we need to here?

File: packages/launchpad/cypress/e2e/experimental-studio.cy.ts
Comment (emilyrohrbough) [inline packages/launchpad/cypress/e2e/experimental-studio.cy.ts:1]: nit: this would need to sync with @ZachJW34's PR but would it make sense for this test file to be named something like, `config-warning.cy.ts` and we can add 
- warns for experimentalStudio
- warns for baseUrl can't connect?

File: packages/data-context/src/data/ProjectLifecycleManager.ts
Comment (emilyrohrbough) [inline packages/data-context/src/data/ProjectLifecycleManager.ts:589]: Can we pull in the types: https://github.com/cypress-io/cypress/blob/083695857dbc4dc3ad7f991cf29967b06523efad/packages/config/lib/validation.d.ts#L2

```suggestion
      (errorKey: string, obj: ErrResponse) => {
```

File: packages/config/lib/index.js
Comment (emilyrohrbough) [inline packages/config/lib/index.js:133]: Can we add a test for this?

**Comment by davidmunechika** [comment]
> According to the 10.0 Changes document, this should throw an error when this is passed: [notion.so/cypressdx/Changes-for-10-0-0-1dca213837b84d6ea297c2d12c615f3a#fc91c34e28054f9b934a1bc6dbbe53a0](https://www.notion.so/cypressdx/Changes-for-10-0-0-1dca213837b84d6ea297c2d12c615f3a#fc91c34e28054f9b934a1bc6dbbe53a0)
> 
> It seems like @brian-mann / @jennifer-shehane / @cowboy have decided this should be a warning? If this is the new path forward, an you update the notion doc?

@emilyrohrbough Yes, I just confirmed with @jennifer-shehane that this will be a warning instead of an error. I will update the notion doc to reflect this.

**Review by emilyrohrbough — DISMISSED** [review]

**Review by ZachJW34 — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]
This previously had 2 :heavy_check_mark: which got tanked by the resolving of merge conflicts. I had another look and the commits since and they were just resolving conflicts. All the tests are :heavy_check_mark: so I'm going to get this one merged up. Thanks @davidmunechika!

**Pull request merged**
